timeline-chart/src/layer/time-graph-chart.ts

```diff
@@ -618,8 +626,27 @@ export class TimeGraphChart extends TimeGraphChartLayer {
             if (el instanceof TimeGraphStateComponent && !this.mousePanning && !this.mouseZooming) {
                 this.selectState(el.model);
             }
-            if (this.mouseInteractions && this.mouseInteractions.click) {
+            
+            var self = this;
+            this._mouseClicks++;
+
+            if (this._mouseClicks === 1) {
+                this._recentlyClickedGlobal = cloneDeep(e.data.global);
+                this._dblClickTimer = setTimeout(function(){
+                    self._mouseClicks = 0;
+                    self._recentlyClickedGlobal = null;
+                }, this._dbClickTime);
+            } else if (this._recentlyClickedGlobal && (this._recentlyClickedGlobal.equals(e.data.global))){
+                clearTimeout(this._dblClickTimer);
+                e.type = 'doubleClick';
+                this._mouseClicks = 0;
+                this._recentlyClickedGlobal = null;
+            }
+
+            if (e.type === 'click' && this.mouseInteractions && this.mouseInteractions.click) {
                 this.mouseInteractions.click(el, e);
+            } else if (e.type === 'doubleClick' && this.mouseInteractions && this.mouseInteractions.doubleClick){
```

Comment:
As mentioned, should not be an if-else, the doubleClick event should be in addition to the click event.